Add Dockerfile for GCP CloudRun FeatureServer

judahrand opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. Ensure that your code follows our code conventions: https://github.com/feast-dev/feast/blob/master/CONTRIBUTING.md#code-style--linting
2. Run unit tests and ensure that they are passing: https://github.com/feast-dev/feast/blob/master/CONTRIBUTING.md#unit-tests
3. If your change introduces any API changes, make sure to update the integration tests scripts here: https://github.com/feast-dev/feast/tree/master/sdk/python/tests or https://github.com/feast-dev/feast/tree/master/sdk/go
4. Make sure documentation is updated for your PR!
5. Make sure you have signed the CLA https://cla.developers.google.com/clas

-->

**What this PR does / why we need it**:
Pretty self explanatory.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes #

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information about release notes, see kubernetes' guide here:
http://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note

```

adchia commented:
/ok-to-test

felixwang9817 commented:
Hey @judahrand, thanks for this PR! We're currently working on serverless deployments on AWS Lambda, so a few things will likely be changing over the next couple of weeks (e.g. in the `app.py` and `config.py` files you have). Are GCP serverless deployments a blocker for you right now? If not, we'd prefer to delay merging in this PR until the AWS Lambda work has landed, which should be in ~2 weeks.

judahrand commented:
Works for me! Happy to rework this then if need be.

felixwang9817 commented:
Great, thanks @judahrand. I'll let you know when the AWS serverless changes have landed.

judahrand commented:
> > @achals Failures don't look like my fault
> 
> yeah i fixed these (integ test clean up issue). Can you remove the role name field from the gcp config? Should be g2g after that.

I did, didn't I? 🤔 Do you mean something other than what I removed?

judahrand commented:
🤷 I just followed the pattern 😛 

I kinda think this is okay. Otherwise, the resources for the Feature Server cloud deployments will get spread out and more difficult to work on.

judahrand commented:
@achals Do you think this is worth merging and then working on the other bits needed to complete the CloudRun support? (The failed test looks like a timeout)
